src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -0,0 +1,355 @@
+
+        input_mapping = {param: arg for param, arg in zip(input_names, input_memlets)}
+        output_mapping = {param.value.data: arg for param, arg in zip(results, output_memlets)}
+        conn_mapping = {param: arg for param, arg in zip(conn_names, conn_memlet)}
+
+        array_mapping = {**input_mapping, **output_mapping, **conn_mapping}
+        symbol_mapping = map_nested_sdfg_symbols(closure_sdfg, context.body, array_mapping)
+
+        nsdfg_node, map_entry, map_exit = self._add_mapped_nested_sdfg(
+            closure_state,
+            sdfg=context.body,
+            map_ranges=map_domain,
+            inputs={**input_mapping, **conn_mapping},
+            outputs=output_mapping,
+            symbol_mapping=symbol_mapping,
+            schedule=dace.ScheduleType.Sequential,
+        )
+        for _, (lb, ub) in closure_domain:
+            map_entry.add_in_connector(lb.value.data)
+            map_entry.add_in_connector(ub.value.data)
+            closure_state.add_edge(
+                lb.value, None, map_entry, lb.value.data, create_memlet_at(lb.value.data, ("0",))
+            )
+            closure_state.add_edge(
+                ub.value, None, map_entry, ub.value.data, create_memlet_at(ub.value.data, ("0",))
+            )
+
+        return closure_sdfg
+
+    def _add_mapped_nested_sdfg(
+        self,
+        state: dace.SDFGState,
+        map_ranges: dict[str, str | dace.subsets.Subset]
+        | list[tuple[str, str | dace.subsets.Subset]],
+        inputs: dict[str, dace.Memlet],
+        outputs: dict[str, dace.Memlet],
+        sdfg: dace.SDFG,
+        symbol_mapping: dict[str, Any] | None = None,
+        schedule: Any = dace.dtypes.ScheduleType.Default,
+        unroll_map: bool = False,
+        location: Any = None,
+        debuginfo: Any = None,
+        input_nodes: dict[str, dace.nodes.AccessNode] | None = None,
+        output_nodes: dict[str, dace.nodes.AccessNode] | None = None,
+    ) -> tuple[dace.nodes.NestedSDFG, dace.nodes.MapEntry, dace.nodes.MapExit]:
+        if not symbol_mapping:
+            symbol_mapping = {sym: sym for sym in sdfg.free_symbols}
+
+        nsdfg_node = state.add_nested_sdfg(
+            sdfg,
+            None,
+            set(inputs.keys()),
+            set(outputs.keys()),
+            symbol_mapping,
+            name=sdfg.name,
+            schedule=schedule,
+            location=location,
+            debuginfo=debuginfo,
+        )
+
+        map_entry, map_exit = state.add_map(
+            f"{sdfg.name}_map", map_ranges, schedule, unroll_map, debuginfo
+        )
+
+        if input_nodes is None:
+            input_nodes = {
+                memlet.data: state.add_access(memlet.data) for name, memlet in inputs.items()
+            }
+        if output_nodes is None:
+            output_nodes = {
+                memlet.data: state.add_access(memlet.data) for name, memlet in outputs.items()
+            }
+        for name, memlet in inputs.items():
```

Comment:
If inputs is empty, we still need an edge to connect the map entry node to the nested sdfg. Same applies to empty outputs and map exit node.

Reply:
fixed